### PR TITLE
[RSDK-9573]   Autodetect height and width of source cam in video store

### PR DESCRIPTION
**Pull request by hexbabe**
[RSDK-9573](https://viam.atlassian.net/browse/RSDK-9573)

[RSDK-9573]: https://viam.atlassian.net/browse/RSDK-9573?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

Tested with config:

```
{
  "components": [
    {
      "name": "webcam-cam",
      "namespace": "rdk",
      "type": "camera",
      "model": "webcam",
      "attributes": {
        "video_path": "FDF90FEB-59E5-4FCF-AABD-DA03C4E19BFB"
      }
    },
    {
      "name": "transform-cam",
      "namespace": "rdk",
      "type": "camera",
      "model": "transform",
      "attributes": {
        "pipeline": [
          {
            "attributes": {
              "y_max_px": 0.4,
              "x_min_px": 0.3,
              "y_min_px": 0.2,
              "x_max_px": 0.6
            },
            "type": "crop"
          },
          {
            "type": "rotate",
            "attributes": {
              "angle_degs": 270
            }
          }
        ],
        "source": "webcam-cam"
      }
    },
    {
      "name": "rtsp-cam-1",
      "namespace": "rdk",
      "type": "camera",
      "model": "viam:viamrtsp:rtsp",
      "attributes": {
        "rtsp_address": "rtsp://localhost:8554/live.stream"
      }
    },
    {
      "name": "video-store-comp",
      "namespace": "rdk",
      "type": "camera",
      "model": "viam:video:storage",
      "attributes": {
        "storage": {
          "segment_seconds": 10,
          "size_gb": 1
        },
        "cam_props": {
          "framerate": 20
        },
        "camera": "rtsp-cam-1",
        "sync": "data"
      },
      "depends_on": [
        "data"
      ]
    }
  ],
  "services": [
    {
      "name": "data",
      "namespace": "rdk",
      "type": "data_manager",
      "attributes": {
        "capture_dir": "/users/seanyu/Downloads/video-store",
        "tags": [],
        "additional_sync_paths": [
          "/users/seanyu/Downloads/video-store"
        ],
        "sync_disabled": false,
        "sync_interval_mins": 0.1
      }
    }
  ],
  "modules": [
    {
      "type": "local",
      "name": "local-viamrtsp-mod",
      "executable_path": "/Users/seanyu/Projects/sean-viamrtsp/bin/darwin-arm64/viamrtsp"
    },
    {
      "type": "local",
      "name": "local-video-store-mod",
      "executable_path": "/Users/seanyu/Projects/sean-video-store/bin/darwin-arm64/video-store"
    }
  ]
}
```

https://github.com/user-attachments/assets/02882b32-56d6-4b32-89ba-af7a7163842e

https://github.com/user-attachments/assets/07694258-3d06-4a3f-872d-fa37ecd7e609

Tested with standard RTSP resolution. Tested also with "weird" resolution using transform cam. See attached vids

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -52,6 +52,9 @@ On the new component panel, copy and paste the following attribute template into
   }
 }
 ```
+
+Additionally, make sure to add your configured data manager service to the `depends_on` array of your `video-store` component.
```
Comment (hexbabe) [inline README.md:56]: TBH we should make data manager an implicit dep using `Validate`

**Comment by seanavery** [comment]
@hexbabe Should we make framerate not required with a default of 20? May need to test out to make sure video still looks ok when framerate is off.

**Comment by hexbabe** [comment]
> @hexbabe Should we make framerate not required with a default of 20? May need to test out to make sure video still looks ok when framerate is off.

Gotcha I can make a quick change and test tomorrow

File: README.md (hunk 1 above)
Comment (seanavery) [reply README.md:56]: Hm yeah that would be nice to reduce setup burden -- I thought we could only do this using builtins?

can we just do something similar to this https://github.com/viamrobotics/rdk/blob/main/components/input/gpio/gpio.go#L49-L59

Comment (hexbabe) [reply README.md:56]: I think I was able to do it in a Python module ?

Comment (hexbabe) [reply README.md:56]: anyways out of scope here

**Comment by hexbabe** [comment]
I also tested to make sure that when we return an error after frame fetch retries reaches the threshold, we keep re-trying to register the model and we don't just give up setting up the module if we can't get a frame. [See](https://viaminc.slack.com/archives/C081S6K8BS4/p1734634568661019)

**Pull request merged**